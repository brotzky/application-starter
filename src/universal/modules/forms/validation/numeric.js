import validator from 'validator';

export default function(value) {
  if (!value) {
    return '';
  }
  
  const valueString = value > -1 ? value.toString() : '';
  return validator.isNumeric(valueString)
    ? undefined
    : {
        defaultMessage: 'can only contain numbers',
        id: 'form.errors.numeric',
        values: {},
      };
}
