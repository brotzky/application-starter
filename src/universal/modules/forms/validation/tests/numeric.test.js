import numeric from '../numeric';

describe('[Form Validation] numeric', () => {
  describe('numeric()', () => {
    it('should return an empty string if nothing is passed it', () => {
      expect(numeric()).toMatchSnapshot();
    });

    it('should return no errors if a string of numbers is passed in', () => {
      expect(numeric('1234')).toMatchSnapshot();
    });

    it('should return an error object when a non-number string is passed in', () => {
      expect(numeric('abc')).toMatchSnapshot();
    });
  });
});
