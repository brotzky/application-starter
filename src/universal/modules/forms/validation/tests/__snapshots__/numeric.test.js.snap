// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Form Validation] numeric numeric() should return an empty string if nothing is passed it 1`] = `""`;

exports[`[Form Validation] numeric numeric() should return an error object when a non-number string is passed in 1`] = `
Object {
  "defaultMessage": "can only contain numbers",
  "id": "form.errors.numeric",
  "values": Object {},
}
`;

exports[`[Form Validation] numeric numeric() should return no errors if a string of numbers is passed in 1`] = `undefined`;
